query-languages/m/m-spec-operators.md

```diff
@@ -619,22 +619,24 @@ The following holds when evaluating an expression containing the relational oper
 
 * Errors raised when evaluating the `x` or `y` operand expressions are propagated.
 
-* The values produced by evaluating both the `x` and `y` expressions must be a number, date, datetime, datetimezone, duration, logical, null or time value. Otherwise, an error with reason code `"Expression.Error"` is raised.
+* The values produced by evaluating both the `x` and `y` expressions must be a binary, date, datetime, datetimezone, duration, logical, number, null, text or time value. Otherwise, an error with reason code `"Expression.Error"` is raised.
 
-* If either or both operands are `null`, the result is the `null` value.
+* Both operands must be of the same type or `null`. Otherwise, an error with reason code `"Expression.Error"` is raised.
```

Comment:
Type equivalence is not defined for arbitrary types, so we cannot meaningfully refer to operands being of the "same" type. Going by the precedent in this document, the best choice here would be to say they must be "the same kind of value". It is also technically correct to say that they have the same kind of _primitive_ type.

Reply:
That is a really good point. Thanks. Tweaked.